lazycache.go

```diff
@@ -9,39 +9,64 @@ import (
 )
 
 type Fetcher func(id string) (interface{}, error)
+type GroupFetcher func() (map[string]interface{}, error)
 
 type Item struct {
 	object  interface{}
 	expires time.Time
 }
 
 type LazyCache struct {
-	fetcher Fetcher
-	ttl     time.Duration
-	lock    sync.RWMutex
-	items   map[string]*Item
+	fetcher      Fetcher
+	groupFetcher GroupFetcher
+	ttl          time.Duration
+	lock         sync.RWMutex
+	items        map[string]*Item
 }
 
 func New(fetcher Fetcher, ttl time.Duration, size int) *LazyCache {
 	return &LazyCache{
-		ttl:     ttl,
-		fetcher: fetcher,
-		items:   make(map[string]*Item, size),
+		ttl:          ttl,
+		fetcher:      fetcher,
+		groupFetcher: nil,
+		items:        make(map[string]*Item, size),
 	}
 }
 
+func NewGroup(groupFetcher GroupFetcher, fetcher Fetcher, ttl time.Duration, size int) *LazyCache {
+	return &LazyCache{
+		ttl:          ttl,
+		fetcher:      fetcher,
+		groupFetcher: groupFetcher,
+		items:        make(map[string]*Item, size),
+	}
+}
+
+func (cache *LazyCache) SwapCache(fetcher Fetcher, groupFetcher GroupFetcher) *LazyCache {
```

Comment:
what is the purpose of this method?

from the name `swap`, I have impression that this method will swap the cache?
like this
```
cache.fetcher = fetcher
cache.groupFetcher = groupFetcher
```

Reply:
For appman rollout, we're using feature flag to swap between using redis and appman api. this will swap the fetch method, but keep the cache (cache is `items  map[string]*Item`) so that requests will still be able to access cache while the switch is happening.

After rollout and removing feature flag, will not need this method anymore